modules/repository-url/src/main/java/org/elasticsearch/plugin/repository/url/URLRepositoryPlugin.java

```diff
@@ -25,13 +24,13 @@
 
 import java.io.IOException;
 import java.util.Arrays;
-import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
 
 public class URLRepositoryPlugin extends Plugin implements RepositoryPlugin {
-    private final SetOnce<URLHttpClient.Factory> httpClientFactory = new SetOnce<>();
+    private final AtomicReference<URLHttpClient.Factory> httpClientFactory = new AtomicReference<>();
```

Comment:
Be careful, this commit will later cause conflicts if things get squashed into the feature branch